File: app/src/main/java/com/google/firebase/sample/cityreviewer/java/viewmodel/MainActivityViewModel.java
package com.google.firebase.sample.cityreviewer.java.viewmodel;

import androidx.lifecycle.ViewModel;

import com.google.firebase.sample.cityreviewer.java.Filters;

/**
 * ViewModel for {@link com.google.firebase.sample.cityreviewer.MainActivity}.
 */

public class MainActivityViewModel extends ViewModel {

    private boolean mIsSigningIn;
    private Filters mFilters;

    public MainActivityViewModel() {
        mIsSigningIn = false;
        mFilters = Filters.getDefault();
    }

    public boolean getIsSigningIn() {
        return mIsSigningIn;
    }

    public void setIsSigningIn(boolean mIsSigningIn) {
        this.mIsSigningIn = mIsSigningIn;
    }

    public Filters getFilters() {
        return mFilters;
    }

    public void setFilters(Filters mFilters) {
        this.mFilters = mFilters;
    }
}

File: app/src/main/java/com/google/firebase/sample/cityreviewer/kotlin/viewmodel/MainActivityViewModel.kt
package com.google.firebase.sample.cityreviewer.kotlin.viewmodel

import androidx.lifecycle.ViewModel
import com.google.firebase.sample.cityreviewer.kotlin.Filters

/**
 * ViewModel for [com.google.firebase.sample.cityreviewer.MainActivity].
 */

class MainActivityViewModel : ViewModel() {

    var isSigningIn: Boolean = false
    var filters: Filters = Filters.default
}
